### ISSUE TITLE
'React/RCTDefines.h' file not found

### ISSUE THREAD

**Issue by tzwickl** [joltup/rn-fetch-blob#461]
Hi, 

Installed Versions:
- "react-native": "0.61.2"
- "rn-fetch-blob": "^0.11.2"

we recently upgraded from react-native version 0.59 to 0.61 and now the `rn-fetch-blob` library no longer works with XCode, we get the following errors when trying to build the project:

<img width="870" alt="Screenshot 2019-10-11 at 5 47 33 PM" src="https://user-images.githubusercontent.com/4968212/66666455-f09ef980-ec50-11e9-9e41-76d32a111f7c.png">

Any ideas or pointers on how we can solve this issue?

Thanks in advance for your support.

Best regards,
Thomas

**Comment by alpha0010** [comment]
Sounds like it might be a library configuration issue.  Are you using the new podfile base autolinking?

**Comment by tzwickl** [comment]
Yes we use the podfile base autolinking. We now downgraded to version 0.60 and now it works :) Seems like it is only an issue with version 0.61 and above

**Comment by romk1n** [comment]
is there a way to fix it for 0.61 ?

**Comment by romk1n** [comment]
actually downgrading to 0.60.6 didn't solve a problem for us :(

**Comment by alpha0010** [comment]
Do native libraries work at all for you?  For example, can you successfully use something like `react-native-webview`?

Also, if you have an old version of Xcode, you may need to update it to at least 10.

**Comment by josectobar** [comment]
I found a work around by adding a condition in the import:
```
#if __has_include("RCTDefines.h")
#import "RCTDefines.h"
#else
#import <React/RCTBridgeModule.h>
#endif
```

**Comment by rcidt** [comment]
I am seeing this as well.

This is being caused by a few of our dependencies which don't support auto linking (they're linked manually).

**Comment by herstand** [comment]
@josectobar Where are you adding that workaround?

**Comment by eyalabadi98** [comment]
Following. We are facingn the same issue  
@josectobar

**Comment by infostreams** [comment]
Ok, I managed to fix it (for me, at least). What you do is the following:

1. Go to Project navigator > Libraries > RNFetchBlob.xcodeproj
2. Scroll down to 'Search Paths'
3. Double click 'Header Search Paths'
4. Add the following entry (click '+'): ```${SRCROOT}/../../../ios/Pods/Headers```.  Make sure to set it to ```recursive```.
5. Clean (cmd + shift + K) and re-build (cmd + B)

Repeat for other libraries that complain about missing ```React/<whatever>.h``` files, perhaps with differing amounts of ```../``` before the ```ios/Pods/Headers/```

**Comment by luatnd** [comment]
> Ok, I managed to fix it (for me, at least). What you do is the following:
> 
> 1. Go to Project navigator > Libraries > RNFetchBlob.xcodeproj
> 2. Click on Build Settings and then the first item under "Targets"
> 3. Scroll down to 'Search Paths'
> 4. Double click to edit the 'Header Search Paths'
> 5. Add the following entry (click '+'): `${SRCROOT}/../../../ios/Pods/Headers`.  Make sure to set it to `recursive`.
> 6. Clean (cmd + shift + K) and re-build (cmd + B)
> 
> Repeat for other libraries that complain about missing `React/<whatever>.h` files, perhaps with differing amounts of `../` before the `ios/Pods/Headers/`

To sum up, I think we should step by step to prevent redundant actions:
1. Check if XCode > Product > Schema > Edit Schema > Build has React(missing) or not
    - If React is present, then you continue to check header search path
    - If React is missing, try to delete it then add it back by pressing Plus sign
         - If you cannot found React in Pods > React after click plus sign. Please check your pod file if someone remove `pod 'React', :path => '../node_modules/react-native/'` from Pod, or they use `target.remove_from_project` to remove it on `post_install`
2. Check headers search path like above quote sections
3. If the issue still exists, it's might be the library *.podspec or sth else, you need to investigate further.

**Comment by judygab** [comment]
> Ok, I managed to fix it (for me, at least). What you do is the following:
> 
> 1. Go to Project navigator > Libraries > RNFetchBlob.xcodeproj
> 2. Click on Build Settings and then the first item under "Targets"
> 3. Scroll down to 'Search Paths'
> 4. Double click to edit the 'Header Search Paths'
> 5. Add the following entry (click '+'): `${SRCROOT}/../../../ios/Pods/Headers`.  Make sure to set it to `recursive`.
> 6. Clean (cmd + shift + K) and re-build (cmd + B)
> 
> Repeat for other libraries that complain about missing `React/<whatever>.h` files, perhaps with differing amounts of `../` before the `ios/Pods/Headers/`

Did anyone got this error while trying this solution?
<img width="366" alt="Screen Shot 2020-03-28 at 12 44 14 PM" src="https://user-images.githubusercontent.com/50160672/77828470-f77fd280-70f1-11ea-8ea9-e1d945ffacae.png">

**Comment by ram95krishh** [comment]
I have the same issue.
React-Native: 0.62.2
React: 16.11.0
rn-fetch-blob: 0.12.0

![image](https://user-images.githubusercontent.com/45874777/81384957-1d0bed00-9130-11ea-9cb6-47f86c62c3f2.png)

![image](https://user-images.githubusercontent.com/45874777/81385008-40cf3300-9130-11ea-8712-e936c8db8b30.png)

Recursive option was also set
![image](https://user-images.githubusercontent.com/45874777/81385026-4a589b00-9130-11ea-9077-cb6a54c2b0e8.png)

**Comment by joeyBerger** [comment]
Having the same issue with react-native-passcode-auth:

    "react": "16.11.0",
    "react-native": "0.62.2",
    "react-native-passcode-auth": "^1.0.0",

**Comment by ravinalamada** [comment]
Hi @judygab, I tried your solution and got the errors you mentioned above too. Did you solve it already? If so, can you please tell me how? Thanks in advance.

**Comment by gokugen** [comment]
Any updates ?

**Comment by saeedtkh** [comment]
any updates ?

**Comment by megancooper** [comment]
After following the `${SRCROOT}/../../../ios/Pods/Headers` I ran into another issue with duplicate symbols. 

~~Anyone have any ideas how to fix this? 🤔~~ (Edit: fix is down below 👇🏾 )

```
❌  duplicate symbol '_OBJC_CLASS_$_RNFetchBlob' in
┌─ librn-fetch-blob.a(RNFetchBlob.o)
└─ libRNFetchBlob.a(RNFetchBlob-a70a61ec5fbdd003117a6dd9c81d14ef.o)

❌  duplicate symbol '_OBJC_METACLASS_$_RNFetchBlob' in
┌─ librn-fetch-blob.a(RNFetchBlob.o)
└─ libRNFetchBlob.a(RNFetchBlob-a70a61ec5fbdd003117a6dd9c81d14ef.o)

❌  ld: 73 duplicate symbols for architecture arm64

❌  clang: error: linker command failed with exit code 1 (use -v to see invocation)

...
```

**Edit (fix found 🚀 ):** Here's the fix:

1. removing `libRNFetchBlob.a` from `Your_Project_Name > General > Frameworks, Libraries, and Embedded Content` (assuming you following [the manual linked guide](https://github.com/joltup/rn-fetch-blob/wiki/Manually-Link-Package#index) like I did)
2. adding the pod directly to my share extension in my `Podfile`:

```
target 'py_project-share-extension' do
  use_react_native!(
    :hermes_enabled => false,
    :flipper_configuration => flipper_config
  )

  pod 'rn-fetch-blob', :path => '../node_modules/rn-fetch-blob'
end
```
3. run `pod install` from the `ios` folder 🤞🏾 


The issue seemed to be that because I was already linking the pods directly (see pic below) I didn't need to manually add the `libRNFetchBlob.a` file. Here's what my final frameworks section looked like:

<img width="759" alt="Screenshot 2023-05-29 at 11 14 51 PM" src="https://github.com/joltup/rn-fetch-blob/assets/8683055/fb8c1fe9-c03a-4445-83da-c89acd00542a">
